arithmetization/src/test/java/net/consensys/linea/zktracer/exceptions/InvalidOpcodeExceptionTest.java

```diff
@@ -37,4 +44,28 @@ void invalidOpcodeExceptionTest() {
         TracedException.INVALID_OPCODE,
         bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
   }
+
+  @ParameterizedTest
+  @MethodSource("nonOpcodeExceptionSource")
+  void nonOpcodeExceptionTest(int value) {
+    BytecodeCompiler program = BytecodeCompiler.newProgram();
+    program.immediate(value);
+    BytecodeRunner bytecodeRunner = BytecodeRunner.of(program.compile());
+    bytecodeRunner.run();
+    assertEquals(
+        TracedException.INVALID_OPCODE,
+        bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+  }
+
+  static Stream<Arguments> nonOpcodeExceptionSource() {
+    List<Arguments> arguments = new ArrayList<>();
+    for (int value = 0; value < 256; value++) {
+      OpCode opCode = OpCode.of(value);
+      // If the value is not a valid opcode, the corresponding value is used in the test
+      if (opCode == OpCode.INVALID) {
+        arguments.add(Arguments.of(value));
+      }
+    }
+    return arguments.stream();
```

Comment:
Won't that just retain precisely the INVALID opcode ?

Reply:
I would expect that you have to go through the Invalid `InstructionFamily`

Reply:
I guess it depends on the implementation of `OpCode.of( ... )`. I did touch related functions recently. I guess it would have to be tested.

Reply:
My bad, it works

<img width="571" alt="image" src="https://github.com/user-attachments/assets/e9e26e9c-92bc-47c1-96bb-5115b113ea84" />

Reply:
Exactly, I replied to your comment on Slack. I'll add a comment to make it clear.

Reply:
I used another approach that makes it more explicit what is happening using:

```
 static Stream<Arguments> nonOpcodeExceptionSource() {
    List<Arguments> arguments = new ArrayList<>();
    for (int value = 0; value < 256; value++) {
      // If value it not in the map, then it is not an OpCode
      if (!valueToOpCodeDataMap.containsKey(value)) {
        arguments.add(Arguments.of(value));
      }
    }
    return arguments.stream();
  }
```